pkg/linter/linter.go

```diff
@@ -315,7 +318,14 @@ func strippedLinter(lctx LinterContext, path string, d fs.DirEntry) error {
 
 	file, err := elf.NewFile(tempfile)
 	if err != nil {
-		// We don't particularly care if this fails, it means it's probably not an ELF file
+		// XXX(Elizafox) - I hate Go's error handling and there's no better way.
+		// It literally gives us nothing but the string. Why Go... WHY...???
```

Comment:
You can check if it's an `elf.FormatError`: https://pkg.go.dev/debug/elf#FormatError

Reply:
The problem with that, is that then I need to check what *kind* of format error. Not all format errors are created equal. A bad magic number can be ignored, but should we ignore bad arch or other issues which could be part of larger issues?

Reply:
I don't follow.  debug/elf does not throw an error for "bad arch".

Reply:
Not arch, class. My bad.

https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/debug/elf/file.go;l=301

Reply:
I think it's always going to be a bit of a heuristic if we're automatically detecting this stuff (rather than having it be explicit in the melange config. E.g. a package might contain a file with some randomly generated data that _happens_ to start with `.ELF` but isn't actually an ELF file, so we might fail with that Class error even though we should just ignore it.

Not sure if we'd rather than false positives (fail on invalid ELF files) or false negatives (ignore invalid ELF files)... probably false positives, right?

Might be better to just do this check yourself here (`Peek` the first 4 bytes with a `bufio.Reader`) rather than calling into `elf.NewFile` and having to decipher the error?

Reply:
I guess I could. I'm not sure if that would be cleaner.